[CI]: eslint failure

### Summary

Recent PRs are failing eslint linter checks with the following: 
```
Run reviewdog/action-eslint@v1
Run $GITHUB_ACTION_PATH/script.sh
🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog
 Running `npm install` to install eslint ...
eslint version:
/home/runner/work/_actions/reviewdog/action-eslint/v1/script.sh: 20: Unknown: not found
 Running eslint with reviewdog 🐶 ...
  /home/runner/work/_actions/reviewdog/action-eslint/v1/script.sh: 23: Unknown: not found
reviewdog: failed to unmarshal rdjson (DiagnosticResult): proto: syntax error (line 1:1): unexpected token 
Error: Process completed with exit code 1.
```

Looks related to this: https://github.com/reviewdog/action-eslint/issues/152

### Proposed fix

The above issues proposes a workaround with:
```
      - uses: actions/setup-node@v3
         with:
           node-version: '18.13'
           cache: 'yarn'
```

However I’m unclear if pinning that version is best practice.

--- a/.github/workflows/reviewdog.yml
+++ b/.github/workflows/reviewdog.yml
@@ -45,6 +45,11 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      # pin to node 18.13 until npm v9 is supported by reviewdog eslint
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.13'
+
       - name: eslint
         uses: reviewdog/action-eslint@v1
         with:

